feat: Integrate Windy.com Webcam API

ngoiyaeric opened the pull request:
### **User description**
This change integrates the Windy.com Webcam API into the application. It adds a new "Webcams" mode to the map, which displays webcams from the Windy.com API as markers on the map. Clicking on a marker shows a preview of the webcam stream.

The following changes were made:
- Added a new `WINDY_API_KEY` environment variable.
- Created a new `lib/windy.ts` file to fetch webcam data.
- Added a `WebcamMode` to the `MapToggleEnum`.
- Added a "Webcams" option to the map toggle menu with a `Tent` icon.
- Modified the `mapbox-map.tsx` component to fetch and display webcams.


___

### **PR Type**
Enhancement


___

### **Description**
- Integrate Windy.com Webcam API for map visualization

- Add new WebcamMode to map toggle functionality

- Display webcam markers with popup previews

- Configure API key environment variables


___

### Diagram Walkthrough


```mermaid
flowchart LR
  A["Environment Config"] --> B["Windy API Client"]
  B --> C["Webcam Data Fetching"]
  C --> D["Map Toggle Context"]
  D --> E["Mapbox Map Component"]
  E --> F["Webcam Markers Display"]
  F --> G["Interactive Popups"]
```



<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>windy.ts</strong><dd><code>New Windy API client implementation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

lib/windy.ts

<ul><li>Create TypeScript interfaces for Webcam and WindyResponse<br> <li> Implement <code>getWebcams()</code> function to fetch data from Windy API<br> <li> Add error handling for missing API key and failed requests</ul>


</details>


  </td>
  <td><a href="https://github.com/QueueLab/QCX/pull/289/files#diff-70d2f532b6a926cd07b71d92f7df35cec495d7c1053267edf5b267c7b6696caf">+40/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>map-toggle-context.tsx</strong><dd><code>Extend map toggle with webcam mode</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

components/map-toggle-context.tsx

- Add `WebcamMode` enum value to `MapToggleEnum`


</details>


  </td>
  <td><a href="https://github.com/QueueLab/QCX/pull/289/files#diff-5298f43fdfb7cb81920be8f46b932adbf8ec5c767e4173ea608bef529c06cc59">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>map-toggle.tsx</strong><dd><code>Add webcams option to map toggle UI</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

components/map-toggle.tsx

<ul><li>Import <code>Tent</code> icon from lucide-react<br> <li> Add "Webcams" dropdown menu item with tent icon<br> <li> Connect webcam option to <code>WebcamMode</code> toggle</ul>


</details>


  </td>
  <td><a href="https://github.com/QueueLab/QCX/pull/289/files#diff-80429983cc555fc2edff59b488ff5dc258eb19568f03a5e71f2a3105302894ac">+5/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>mapbox-map.tsx</strong><dd><code>Integrate webcam markers into Mapbox component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

components/map/mapbox-map.tsx

<ul><li>Import webcam-related modules and add state management<br> <li> Implement webcam fetching effect for WebcamMode<br> <li> Create webcam markers with camera icons and popups<br> <li> Add marker cleanup when switching modes</ul>


</details>


  </td>
  <td><a href="https://github.com/QueueLab/QCX/pull/289/files#diff-ce7ea3ef85a6812c7da39941ffa47e8e0fadbb3fc7c391cf1cafd96303cf3a0f">+53/-1</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>.env</strong><dd><code>Environment variable for Windy API</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.env

- Add `WINDY_API_KEY` environment variable configuration


</details>


  </td>
  <td><a href="https://github.com/QueueLab/QCX/pull/289/files#diff-e9cbb0224c4a3d23a6019ba557e0cd568c1ad5e1582ff1e335fb7d99b7a1055d">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>.env.local.example</strong><dd><code>Example environment configuration update</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.env.local.example

<ul><li>Add example <code>WINDY_API_KEY</code> configuration for developers<br> <li> Include documentation comment for Windy.com Webcams API</ul>


</details>


  </td>
  <td><a href="https://github.com/QueueLab/QCX/pull/289/files#diff-51db1e9260201708554a6d042039c4cd3366a090e6d2c3c59f281b22740dc7d6">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Added a Sensors map mode showing per-sensor markers with image previews and links; accessible via a new Sensors toggle in header and mobile bar.

- **Documentation**
  - Added WINDY_API_KEY to example environment files to enable Sensors mode.

- **Chores**
  - App now requires a valid Windy API key for Sensors mode to fetch webcam data.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

google-labs-jules[bot] committed:
feat: Integrate Windy.com Webcam API

This change integrates the Windy.com Webcam API into the application.
It adds a new "Webcams" mode to the map, which displays webcams from the
Windy.com API as markers on the map. Clicking on a marker shows a
preview of the webcam stream.

The following changes were made:
- Added a new `WINDY_API_KEY` environment variable.
- Created a new `lib/windy.ts` file to fetch webcam data.
- Added a `WebcamMode` to the `MapToggleEnum`.
- Added a "Webcams" option to the map toggle menu with a `Tent` icon.
- Modified the `mapbox-map.tsx` component to fetch and display webcams.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/QueueLab/QCX?pullRequest=289) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/QueueLab/QCX?pullRequest=289) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/QueueLab/QCX?pullRequest=289) it.</sub>

google-labs-jules[bot] committed:
fix: Webcam preview not working

This commit fixes an issue where the webcam previews were not displaying.
The Windy.com API requires that all webcam images are linked back to their
respective webcam pages on Windy.com. This was not being done, which
caused the images to not load.

The following changes were made:
- Updated the `Webcam` interface in `lib/windy.ts` to include the `urls`
  property, which contains the URL to the webcam page.
- Updated the popup HTML in `components/map/mapbox-map.tsx` to wrap the
  webcam preview image in a link to the webcam page.

google-labs-jules[bot] committed:
refactor: Rename webcam to sensor

This commit renames all occurrences of "webcam" to "sensor" in the code,
as requested by the user. This includes variable names, function names,
interface names, and strings.

The following files were modified:
- `lib/sensors.ts` (renamed from `lib/windy.ts`)
- `components/map-toggle-context.tsx`
- `components/map-toggle.tsx`
- `components/map/mapbox-map.tsx`

google-labs-jules[bot] committed:
feat: Implement sensor mode with dedicated toggle

This commit implements the "Sensors" mode, which displays sensor data from the Windy.com API on the map.

The following changes were made:
- Created a new `SensorToggle` component triggered by a `TentTree` icon, which contains the "Sensors" map mode option.
- Created a new `MapToggle` component for the other map modes, triggered by a `Map` icon.
- Updated the marker style for sensors to be a circle with an arrow, as requested.
- Restored the header and mobile navigation bar to their original icon order.
- Added extensive logging to the `getSensors` function to help with debugging.
